File: ParseWeight/ParseWeight.m
%%
clc; close all; clear all;

%% Load weight values
load 'conv0.mat';
load 'conv1.mat';
load 'conv2.mat';

%%
conv_size = size(paW0);
f_conv = fopen('conv0.txt','w');
for h = 1:conv_size(1,3)
    for k = 1:conv_size(1,4)
        for i=1:conv_size(1,1)
            for j=1:conv_size(1,2)
                fprintf(f_conv,'%0.5f\n',paW0(i,j,h,k));
            end
        end
    end
end
fclose(f_conv);

conv_size = size(paW6);
f_conv = fopen('conv1.txt','w');
for h = 1:conv_size(1,3)
    for k = 1:conv_size(1,4)
        for i=1:conv_size(1,1)
            for j=1:conv_size(1,2)
                fprintf(f_conv,'%0.5f\n',paW6(i,j,h,k));
            end
        end   
    end
end
fclose(f_conv);

conv_size = size(paW12);
f_conv = fopen('conv2.txt','w');
for h = 1:conv_size(1,3)
    for k = 1:conv_size(1,4)
        for i=1:conv_size(1,1)
            for j=1:conv_size(1,2)
                fprintf(f_conv,'%0.5f\n',paW12(i,j,h,k));
            end
        end
    end
end
fclose(f_conv);
        

File: ParseWeight/ParseWeight.py
#!/usr/bin/env python

#Read trained networks' configurations
#Author: Nghia Doan, nghia.doan@mail.mcgill.ca
#Date  : 2017, Nov 29, 11:14 AM

#%%
import numpy as np
import scipy.io

#%%
# weight file
dir = './nghia_CNN.npz'
print('Read from file: ' + dir +'\n')
npzFile = np.load(dir)
print ('All classes: ') 
print (npzFile.files) 
print ('\n')

#%%
# needed files
idx = 3
paW = npzFile['pareto_weights']
paW = paW[idx-1]
paW0 = paW[0]
paW6 = paW[6]
paW12 = paW[12]

#%%
scipy.io.savemat('./conv0.mat', mdict={'paW0': paW0})
scipy.io.savemat('./conv1.mat', mdict={'paW6': paW6})
scipy.io.savemat('./conv2.mat', mdict={'paW12': paW12})
#%%
